source/fundamentals/connection/jndi.txt

```diff
@@ -0,0 +1,100 @@
+         ``module.xml`` file into the module:
+
+         .. literalinclude:: /includes/fundamentals/code-snippets/wildfly-module-with-version.xml
+            :language: xml
+
+
+      #. Add a binding to the naming subsystem configuration that references the
+         above module, the ``MongoClientFactory`` class, and the
+         :ref:`connection string <connection-uri>` for the MongoDB cluster.
+
+         .. code-block:: xml
+
+            <subsystem xmlns="urn:jboss:domain:naming:2.0">
+               <bindings>
+                  <object-factory name="java:global/MyMongoClient" module="org.mongodb" class="com.mongodb.client.MongoClientFactory">
+                     <environment>
+                        <property name="connectionString" value="mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority"/>
+                     </environment>
+                  </object-factory>
+               </bindings>
+               <remote-naming/>
+            </subsystem>
+
+         .. note::
+
+            Replace the placeholder connection value URI in the ``property`` tag
+            with a value that points to your MongoDB installation.
+
+      This will make a MongoClient instance accessible via the JNDI name
+      ``MyMongoClient`` in the ``java:global`` context.
+
+   .. tab:: Tomcat
+      :tabid: tomcat
+
+      1. Copy the ``mongo-java-driver.jar`` jar file into the ``lib`` directory
+         of your `Tomcat <http://tomcat.apache.org/>`__ installation.
+
+      #. In ``context.xml`` of your application, add a resource that references
+         the ``MongoClientFactory`` class and the :ref:`connection string
+         <connection-uri>` for the MongoDB cluster:
+
+         .. code-block:: xml
+
+            <Resource name="mongodb/MyMongoClient"
+               auth="Container"
+               type="com.mongodb.MongoClient"
+               closeMethod="close"
+               factory="com.mongodb.client.MongoClientFactory"
+               singleton="true"
+               connectionString="mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority"/>
+
+         .. note::
+
+            Replace the placeholder connection value URI in the ``connectionString``
+            attribute with a value that points to your MongoDB installation.
+
+      #. In ``web.xml`` of your application, add a reference to the
+         ``MongoClientFactory`` resource defined in the previous step:
+
+         .. code-block:: xml
+
+            <resource-ref>
+              <res-ref-name>
+                mongodb/MyMongoClient
+              </res-ref-name>
+              <res-type>
+                com.mongodb.MongoClient
+              </res-type>
+              <res-auth>
+                Container
+              </res-auth>
+            </resource-ref>
+
+      This will make a ``MongoClient`` instance accessible via the JNDI name
+      ``mongodb/MyMongoClient`` in the ``java:comp/env`` context.
```

Comment:
@jyemin do you think its worth adding a note that MongoClientFactory's are simple wrappers and for any non support JNDI implementations users can easily create their own factory?

Reply:
Open to this, if @jyemin thinks it's a good idea.

Reply:
Yes, it's a good idea.  Though I haven't heard anyone ask about JNDI in ages.